src/db.c

```diff
@@ -1049,15 +1119,51 @@ void scanGenericCommand(client *c, robj *o, unsigned long cursor) {
     listRelease(keys);
 }
 
+void addSlotIdToCursor(int slot, unsigned long long int *cursor) {
+    if (slot > 0) {
```

Comment:
nit: can we use db->dict_count > 1?
I know current impl also works but I think using `db->dict_count` could be a better indicator.

Reply:
Explained motivation in the comment above.

Reply:
Can we add a comment indicating that slot could be -1 if we have no slot left? That would help people to understand the `slot > 1` condition check

Reply:
Absolutely.